### PR TITLE
Add comment form and function that handles a POST request from the form. 

### PR DESCRIPTION
**Pull request by AnudeepYakkala**
Added a form with a text input that allows users to enter a comment. Once the submit button is pressed, a POST request is made to "/data". 

Added a function on the back-end that receives this POST request. It adds the text from the comment form to the comments ArrayList in the backed. Then it redirects back to the comments section of the home page. 

When the homepage is reloaded, the getMessage() function is called which adds all the stored comments to the page. 

The following screenshot shows the changes:
image

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,9 +25,52 @@
...
+    return json;
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```
Comment (officialcwang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: Is it possible to use === instead here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,9 +25,52 @@
...
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> messages = new ArrayList<>();
+
+  /**
+   * Write to /data the messages ArrayList as a json string
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = convertToJson(messages);
+
```
Comment (officialcwang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:36]: nit: Remove unnecessary space here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -23,9 +25,52 @@
...
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Obtain the input from the comment form and add it to the messages ArrayList
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment input from the form.
+    String comment = getParameter(request, "text-input", "");
```
Comment (officialcwang) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: nit: Add parameter name comment to "" 
https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: Oops, forgot that you need 3 equals in JavaScript.

Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: I think == should be fine here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: Okay.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -25,19 +25,51 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> messages = new ArrayList<>(Arrays.asList("Hello", "Bonjur", "Hola"));
+  private ArrayList<String> messages = new ArrayList<>();
 
+  /**
+   * Write to /data the messages ArrayList as a json string
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: Add period at end of line.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -25,19 +25,51 @@
...
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println(convertToJson(messages));
+    response.sendRedirect("index.html#comments-container");
+  }
+
+  /**
+   * @return the ArrayList paramater converted to a json format
+   */
   public String convertToJson(ArrayList<String> messages) {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: convertMessageListToJson? Just make the functionality a bit more explicit.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -25,19 +25,51 @@
...
     Gson gson = new Gson();
     String json = gson.toJson(messages);
     return json;
   }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: Prefer to use an optional for this function: 
`Optional<String> getParameter(HttpServletRequest request, String param) `
Effective Java Item 55: Return optionals judiciously

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Sure!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: Okay, I will change it.

**Pull request merged**